schema/schema_test.go

```diff
@@ -0,0 +1,389 @@
+package schema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCollectionSchema(t *testing.T) {
+	// invalid
+	type empty struct {
+	}
+
+	// invalid
+	type noPK struct {
+		Key1 string `json:"key_1"`
+	}
+
+	// valid. implicit one part primary key with index 1
+	type pk struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key"`
+	}
+
+	// invalid. index starts from 1
+	type pk0 struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key:0"`
```

Comment:
why not start with 0?

Reply:
I think 1 is more natural for index parts. Also GORM indexes start from 1, so people are used to it.

Reply:
So it is starting from 1 here and then in Java, I think we are starting with 0. Can we make both language behavior consistent?

Reply:
Same, starting from 1.